tracing-appender/src/rolling.rs

```diff
@@ -485,6 +530,18 @@ impl Rotation {
                     .expect("Invalid time; this is a bug in tracing-appender");
                 date.replace_time(time)
             }
+            Rotation::WEEKLY => {
+                let time = Time::from_hms(0, 0, 0)
+                    .expect("Invalid time; this is a bug in tracing-appender");
+                let date = date.replace_time(time);
+                let rounded_ordinal = Rotation::round_ordinal_to_week(date.date());
+
+                date.replace_date(
+                    Date::from_ordinal_date(date.year(), rounded_ordinal).expect(
+                        "ordinal should never be out of range; this is a bug in tracing-appender",
+                    ),
+                )
```

Comment:
I feel that this approach may be surprising to some users. It means that this year, the files will rotate on a Wednesday morning (UTC), but last year they would have rotated on a Monday.

Instead, I would suggest that we rotate on a fixed day of the week always. My preference would be on Sunday, because that's the day that `cron` uses for 0. This may not be what everyone wants either, but without providing a configurable option, we just have to pick a day.

We should also call out explicitly in the docs for `weekly()` when the rotation will occur, since I don't think that any option we choose will be obvious to everyone.

Reply:
Thanks for the review, Hayden!

I've given this some thought and I agree with you. For my particular use case, I didn't care as much about the consistency, but I agree that this is probably going to be unexpected for users.

I really like the idea of matching `cron` behavior here. I'll work on those changes soon and get back to you.

Thanks again!